src/java/org/apache/cassandra/db/compaction/SharedCompactionObserver.java

```diff
@@ -34,16 +39,25 @@
 /// if assertions are enabled.
 public class SharedCompactionObserver implements CompactionObserver
 {
+    private static final Logger logger = LoggerFactory.getLogger(SharedCompactionObserver.class);
+
     private final AtomicInteger toReportOnComplete = new AtomicInteger(0);
     private final AtomicReference<Throwable> onCompleteException = new AtomicReference(null);
     private final AtomicReference<CompactionProgress> inProgressReported = new AtomicReference<>(null);
     private final CompactionObserver observer;
```

Comment:
Let's drop the special status of this observer, and put it in the observers' list. This will also simplify the signaling code.